pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -366,7 +366,7 @@ def extra(self) -> Dict[str, Any]:
         return self.get('_extra', {})
 
     @extra.setter
-    def extra(self, extra: Mapping[str, Any]) -> None:
+    def extra(self, extra: Dict[str, Any]) -> None:
```

Comment:
why must it be a dict and cannot be any mapping? Is it just because it must be mutable? then we could use `MutableMapping` I think.

Reply:
> Is it just because it must be mutable?

Yes

>  then we could use MutableMapping I think.

Sure, I can change it. What's the difference?

Reply:
I think MutableMapping is not as restrictive ( a dict is a special case of a muable mapping). But I'm fine with Dict as well (in fact that's the only mutable mapping people commonly use in practice, I think).